browser-interface/packages/shared/profiles/retrieveProfile.ts

```diff
@@ -0,0 +1,25 @@
+import type { Avatar } from '@dcl/schemas'
+import { store } from 'shared/store/isolatedStore'
+import { sagaMiddleware } from 'shared/store/store'
+import { profileRequest } from './actions'
+import { fetchProfile } from './sagas/fetchProfile'
+import { profile } from './selectors'
+
+/**
+ * Retrieves a Profile with a version equal or higher than the provided one (if any)
+ * - If the `userId` is the current user, look in local storage
+ * - Otherwise, tries to fetch it from the in-memory redux store registry of profiles
+ * - Next, tries to fetch it from comms using an ad-hoc "RPC"
+ *   * if userId is a connected peer, send a request to fetch the profile from them
+ * - Finally, look for a deployed profile on the catalyst
+ */
+export async function retrieveProfile(userId: string, minimumVersion?: number): Promise<Avatar> {
```

Comment:
here it is. thanks for the comments, do you think it makes sense to unify that `fetchProfile` saga in this file?

Reply:
because `fetchProfile` is a saga, I wanted to keep it in the `sagas` folder -- any ideas on how could we enforce "use this one function and not the other!"?